### PR TITLE
05875 query workflow improvements

### PR DESCRIPTION
**Pull request by netopyr**
Simplifications of the query workflow that I already made available. Had to redo them, because squash&merge does not work well with switching base branches.

- Removed `QueryContext`
- Removed metrics in `QueryWorkflowImpl`
- Introduced `NodeInfo`, `NetworkInfo`, and `CurrentPlatformStatus` in hedera-app and provided mono adapters
- Some minor simplifications in `QueryWorkflowImpl`

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java:25]: FunctionalInterface

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:25]: FunctionalInterface

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:32]: NotNull annotation

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:27]: FunctionalInterface

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java
Comment (hendrikebbers) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:34]: NotNull annotation

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java:35]: nonnulla annotations + nullcheck

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java:28]: nonnulla annotations + nullcheck

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNodeInfo.java:34]: nonnulla annotations + nullcheck

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java
Comment (hendrikebbers) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:227]:
```suggestion
            throw new StatusRuntimeException(Status.INVALID_ARGUMENT, e);
```

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java
Comment (hendrikebbers) [inline hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:172]: networkInfo can be null

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java:32]: Fixed

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/CurrentPlatformStatus.java:25]: It is the first version of a facility which will get more methods later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:25]: It is the first version of a facility which will get more methods later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NetworkInfo.java:32]: Fixed

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:27]: It is the first version of a facility which will get more methods later.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/info/NodeInfo.java:34]: I see a pattern. Is Chat GPT making your code reviews? 🙂

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoCurrentPlatformStatus.java:35]: Fixed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/MonoNetworkInfo.java:28]: Fixed

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java
Comment (netopyr) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java:227]: `StatusRuntimeException` does not allow this.

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java
Comment (netopyr) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusGetTopicInfoHandler.java:172]: It should not. A `networkInfo` always has to be provided during construction. The empty constructor is just left in there for now to make the project compile.

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by jeromy-cannon — APPROVED** [review]
from a release engineering perspective I approve the build changes, please wait for other approvers for remaining code before merging.

**Pull request merged**
